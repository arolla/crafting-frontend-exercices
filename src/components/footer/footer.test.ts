import Footer from './footer'

describe('Components', () => {
  describe('Footer', () => {
    beforeAll(() => {
      Footer.register()
    })
    test('should provide a "Log in" link given user is NOT logged in', () => {
      // Given
      const footer = new Footer()
      footer.render()
      // When
      footer.setAttribute('is-user-logged-in', 'false')
      // Then
      expect(footer.shadowRoot).toBeTruthy()
      expect(footer.shadowRoot?.querySelector('#footer')).toMatchSnapshot()
    })
    test('should provide a "Log out" link given user is logged in', () => {
      // Given
      const footer = new Footer()
      footer.render()
      // When
      footer.setAttribute('is-user-logged-in', 'true')
      // Then
      expect(footer.shadowRoot).toBeTruthy()
      expect(footer.shadowRoot?.querySelector('#footer')).toMatchSnapshot()
    })
  })
})
