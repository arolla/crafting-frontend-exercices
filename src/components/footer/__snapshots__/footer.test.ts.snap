// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Footer should provide a "Log in" link given user is NOT logged in 1`] = `
<div
  id="footer"
>
  
  
  <h3>
    Need help?
  </h3>
  
  
  <p>
    Contact | Map | Log in
  </p>
  

</div>
`;

exports[`Components Footer should provide a "Log out" link given user is logged in 1`] = `
<div
  id="footer"
>
  
  
  <h3>
    Need help?
  </h3>
  
  
  <p>
    Contact | Map | Log out
  </p>
  

</div>
`;
